### PR TITLE
fix: updated unit and integration tests in release pipeline

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- updated unit tests in release pipeline
- added integration tests in release pipeline

**Related issue(s)**

- https://github.com/kyma-project/eventing-auth-manager/issues/149

### CODE REVIEW

File: .github/workflows/create-release.yml [hunk 1]
```diff
@@ -72,7 +72,51 @@ jobs:
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: main
+
+      - name: Cache Go dependencies
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.21'
```
Comment (marcobebway) [inline .github/workflows/create-release.yml:85]: Why not `1.22`?

File: .github/workflows/create-release.yml [hunk 2]
```diff
@@ -72,7 +72,51 @@ jobs:
...
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          ref: main
+
+      - name: Setup Golang
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.21'
```
Comment (marcobebway) [inline .github/workflows/create-release.yml:103]: Why not `1.22`?

File: .github/workflows/create-release.yml (hunk 1 above)
Comment (mfaizanse) [reply .github/workflows/create-release.yml:85]: The [PR jobs](https://github.com/kyma-project/eventing-auth-manager/blob/main/.github/workflows/unit-test.yml) are still using 1.21 so just used the same version for now. Later we can bump them together.

File: .github/workflows/create-release.yml (hunk 2 above)
Comment (mfaizanse) [reply .github/workflows/create-release.yml:103]: The [PR jobs](https://github.com/kyma-project/eventing-auth-manager/blob/main/.github/workflows/unit-test.yml) are still using 1.21 so just used the same version for now. Later we can bump them together.

**Review by marcobebway — APPROVED** [review]

**Pull request merged**